Assignment4/PauVL-d21c024/task07.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+"""Copia de Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1P8fNozqnrmPc2GAzQdtx9ooZadN3CTPu
+
+**Task 07: Querying RDF(s)**
+"""
+
+#!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+print("List all subclasses of \"Person\" with RDFLib and SPARQL")
+
+# Aunque se pedía sacar solo el primer nivel de subclase, yo he listado además todas las subclasaes de Person indepentemente de su nivel de subclase.
+
+from rdflib.plugins.sparql import prepareQuery
+NS = Namespace("http://somewhere#")
+q1 = prepareQuery('''
+  SELECT ?Subclase WHERE { 
+    ?Subclase rdfs:subClassOf+ ns:Person. 
+  }
+  ''',
+  initNs = { "rdfs": RDFS, "ns": NS}
+)
+print("7.1")
+for r in g.query(q1):
+  print(r)
+print("FIN")
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+print("List all individuals of \"Person\" with RDFLib and SPARQL (remember the subClasses)")
+VCARD = Namespace("http://somewhere#http://www.w3.org/2001/vcard-rdf/3.0/")
+q2 = prepareQuery('''
+  SELECT ?X WHERE {
+    ?Subclase rdfs:subClassOf* ns:Person. 
+    ?X rdf:type ?Subclase.
+  }
+  ''',
+  initNs = { "rdfs": RDFS, "ns": NS, "vcard": VCARD, "rdf": RDF,}
+)
+
+print("7.2")
+for r in g.query(q2):
+  print(r.X)
+print("FIN")
+
+"""**TASK 7.3: List all individuals of "Person" and all their properties including their class with RDFLib and SPARQL**
+
+"""
+
+print("List all individuals of \"Person\" and all their properties including their class with RDFLib and SPARQL")
+q3 = prepareQuery('''
+  SELECT ?X ?Subclase ?A ?B ?C WHERE {
+    ?Subclase rdfs:subClassOf* ns:Person. 
+    ?X rdf:type ?Subclase.
+    ?X vcard:FN ?A.
+    ?X vcard:Given ?B.
+    ?X vcard:Family ?C.
+  }
+  ''',
+  initNs = {"rdf": RDF, "ns": NS, "vcard": VCARD}
+)
+
+print("7.3")
+for r in g.query(q3):
+  print(r)
+print("FIN")
```

Comment:
Missing 7.1 .. 7.3 in rdflib

Reply:
Ya he modificado el Assignment 2b y he corregido lo del namespace en el task06 del Assignment 4; pero no se a que te refieres cuando dices que falta el task07, así que lo he vuelto a subir (el task07) poniendo al final de cada apartado print("FIN 7.elapartado") en vez de solo print("FIN") como estaba antes.